draft-ietf-taps-impl.md

```diff
@@ -123,7 +123,7 @@ The connection objects that are exposed to applications for Transport Services a
 
 Preconnection objects should be implemented as bundles of properties that an application can both read and write. A Preconnection object influences a Connection only at one point in time: when the Connection is created. Connection objects represent the interface between the application and the implementation to manage transport state, and conduct data transfer. During the process of establishment ({{conn-establish}}), the Connection will not be bound to a specific transport protocol instance, since multiple candidate Protocol Stacks might be raced. 
 
-Once a Preconnection has been used to create an outbound Connection or a Listener, the implementation should ensure that the copy of the properties held by the Connection or Listener is not affected when the application makes changes to a Preconnection object. This may involve the implementation performing a deep-copy, copying the object with all the objects that it references.
+Once a Preconnection has been used to create an outbound Connection or a Listener, the implementation should ensure that the copy of the properties held by the Connection or Listener cannot be mutated by the application makes changes to the original Preconnection object. This may involve the implementation performing a deep-copy, copying the object with all the objects that it references, or making the Preconnection object immutable.
```

Comment:
I do not see how the last addition solves the contradiction complained about by @britram. It seems to make it worse? But I also do not see the problem with the original text. maybe the addition of "the original" is enough?

Nit: makes --> making.

Reply:
```suggestion
Once a Preconnection has been used to create an outbound Connection or a Listener, the implementation should ensure that the copy of the properties held by the Connection or Listener cannot be mutated by the application makes changes to the original Preconnection object. This may involve the implementation performing a deep-copy, copying the object with all the objects that it references.
```
```suggestion
Once a Preconnection has been used to create an outbound Connection or a Listener, the implementation should ensure that the copy of the properties held by the Connection or Listener cannot be mutated by the application makes changes to the original Preconnection object. This may involve the implementation performing a deep-copy, copying the object with all the objects that it references, or making the Preconnection object immutable.
```